.github/workflows/common/reset-self-hosted-runner/action.yml

```diff
@@ -0,0 +1,22 @@
+name: Reset Self-hosted Runner
+description: Shared action that allows to reset a runner to its original state
+runs:
+  using: "composite"
+  steps:
+    - name: Remove Cargo Generated Artifacts
+      shell: bash
+      run: |
+        echo "Removing generated cargo artifacts..."
+        rm -fr ./target
```

Comment:
Do we want this to just use `cargo clean`? I guess we don't want to have to have the cargo dep at this point however.